modules/rostests/kmtests/ntos_ob/ObInsert.c

```diff
@@ -0,0 +1,651 @@
+/*
+ * PROJECT:         ReactOS kernel-mode tests
+ * LICENSE:         GPLv2+ - See COPYING in the top level directory
+ * PURPOSE:         Kernel-Mode Test Suite - ObInsertObject test
+ * PROGRAMMER:      Gleb Surikov <glebs.surikovs@gmail.com>
+ */
+
+#include <kmt_test.h>
+
+#define NDEBUG
+#include <debug.h>
+
+typedef struct DUMMY_TYPE
+{
+    ULONG Data;
+} DUMMY_TYPE, *PDUMMY_TYPE;
+
+static HANDLE DirectoryHandle;
+static UNICODE_STRING DirectoryName;
+
+static POBJECT_TYPE DummyType;
+static UNICODE_STRING DummyTypeName;
+static OBJECT_TYPE_INITIALIZER ObjectTypeInitializer;
+static OBJECT_ATTRIBUTES DummyObjectAttributes;
+
+static
+VOID
+NTAPI
+DumpProc(
+    IN PVOID Object,
+    IN POB_DUMP_CONTROL DumpControl
```

Comment:
Since you write new code, please use SAL2 annotations instead of these old-school things:
Here's a little dictionary for the simplest ones:

| Old macro annotation | New SAL2 annotation |
| :---: | :---: |
| `IN` | `_In_` |
| `OUT` | `_Out_` |
| `IN OUT` | `_Inout_` |
| `IN whatever OPTIONAL` | `_In_opt_ whatever` |
| `OUT whatever OPTIONAL` | `_Out_opt_ whatever` |
| `IN OUT whatever OPTIONAL` | `_Inout_opt_ whatver` |

Reply:
Thanks!